services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/utils/RestPreconditionsUtil.java

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+
+package com.futurewei.alcor.subnet.utils;
+
+import com.futurewei.alcor.common.entity.CustomerResource;
+import com.futurewei.alcor.common.exception.*;
+import com.futurewei.alcor.subnet.entity.VpcState;
+import org.thymeleaf.util.StringUtils;
+
+public class RestPreconditionsUtil {
+    public static <T> T verifyResourceFound(T resource) throws ResourceNotFoundException {
```

Comment:
Since this file is not SubnetManager specific, let us use the one under /lib.

Reply:
Okay ~ I'll move this utils to lib